presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorTableHandle.java

```diff
@@ -15,4 +15,8 @@
 
 public interface ConnectorTableHandle
 {
+    /**
+     * Allow the connector to reject the table scan produced by the planner.
+     */
+    default void validateScan(ConnectorSession session) {}
```

Comment:
This should be implemented as a ConnectorMetadata API. Handles are meant to be data objects, not service APIs (think of them as some kind of opaque ID)